### DISCUSSION TITLE
why cgo is not supported?

### DISCUSSION THREAD

**Discussion by nikitos1550** [u-root/u-root#2757] [Q&A] [answered]
Hello! What is main reason why cgo is not supported?

**Comment by hugelgupf** [comment, marked as answer]
It's possible that cgo could work, but the way gobusybox works involves some rewriting of source, and creating a bunch of stuff in temporary directories [as described here](https://github.com/u-root/gobusybox?tab=readme-ov-file#how-it-works). It'd presumably break cgo compilations.

We don't plan to support it ourselves as the mixing of Go and C using cgo bloats binary size (e.g. not that easy to do dead code elimination). Downstream users that need to use C in their initramfs images tend to build a small, statically linked C binary and call it from Go instead.
